internal/engine/compiler/engine.go

```diff
@@ -844,7 +857,13 @@ func (ce *callEngine) deferredOnCall(recovered interface{}) (err error) {
 				break
 			}
 		}
+
 		err = builder.FromRecovered(recovered)
+		// Call listeners in revers order since they were captured while
+		// rewinding the call stack.
+		for i := len(functionListeners) - 1; i >= 0; i-- {
+			functionListeners[i].Abort(ctx, m, functionListeners[i].def, err)
```

Comment:
this was Abort being called, which I missed. I'll re-do review as this changes things!